radiation/python/radlw/taumol_gt4py.py

```diff
@@ -0,0 +1,4957 @@
+)
+def taugb01(
+    pavel: FIELD_FLT,
+    colamt: Field[type_maxgas],
+    colbrd: FIELD_FLT,
+    fac00: FIELD_FLT,
+    fac01: FIELD_FLT,
+    fac10: FIELD_FLT,
+    fac11: FIELD_FLT,
+    jp: FIELD_INT,
+    jt: FIELD_INT,
+    jt1: FIELD_INT,
+    selffac: FIELD_FLT,
+    selffrac: FIELD_FLT,
+    indself: FIELD_INT,
+    forfac: FIELD_FLT,
+    forfrac: FIELD_FLT,
+    indfor: FIELD_INT,
+    minorfrac: FIELD_FLT,
+    scaleminorn2: FIELD_FLT,
+    indminor: FIELD_INT,
+    fracs: Field[type_ngptlw],
+    taug: Field[type_ngptlw],
+    absa: Field[(DTYPE_FLT, (10, 65))],
+    absb: Field[(DTYPE_FLT, (10, 235))],
+    selfref: Field[(DTYPE_FLT, (10, 10))],
+    forref: Field[(DTYPE_FLT, (10, 4))],
+    fracrefa: Field[(DTYPE_FLT, (10,))],
+    fracrefb: Field[(DTYPE_FLT, (10,))],
+    ka_mn2: Field[(DTYPE_FLT, (10, 19))],
+    kb_mn2: Field[(DTYPE_FLT, (10, 19))],
+    ind0: FIELD_INT,
+    ind0p: FIELD_INT,
+    ind1: FIELD_INT,
+    ind1p: FIELD_INT,
+    inds: FIELD_INT,
+    indsp: FIELD_INT,
+    indf: FIELD_INT,
+    indfp: FIELD_INT,
+    indm: FIELD_INT,
+    indmp: FIELD_INT,
+    pp: FIELD_FLT,
+    corradj: FIELD_FLT,
+    scalen2: FIELD_FLT,
+    tauself: FIELD_FLT,
+    taufor: FIELD_FLT,
+    taun2: FIELD_FLT,
+):
+    from __externals__ import nspa, nspb, laytrop, ng01, nlay
+
+    with computation(PARALLEL), interval(0, laytrop):
+        ind0 = ((jp - 1) * 5 + (jt - 1)) * nspa
+        ind1 = (jp * 5 + (jt1 - 1)) * nspa
+        inds = indself - 1
+        indf = indfor - 1
+        indm = indminor - 1
+
+        ind0p = ind0 + 1
+        ind1p = ind1 + 1
+        indsp = inds + 1
+        indfp = indf + 1
+        indmp = indm + 1
+
+        pp = pavel
+        scalen2 = colbrd * scaleminorn2
+        if pp < 250.0:
+            corradj = 1.0 - 0.15 * (250.0 - pp) / 154.4
+        else:
+            corradj = 1.0
+
+        for ig in range(ng01):
+            tauself = selffac * (
+                selfref[0, 0, 0][ig, inds]
+                + selffrac * (selfref[0, 0, 0][ig, indsp] - selfref[0, 0, 0][ig, inds])
+            )
+            taufor = forfac * (
+                forref[0, 0, 0][ig, indf]
+                + forfrac * (forref[0, 0, 0][ig, indfp] - forref[0, 0, 0][ig, indf])
+            )
+            taun2 = scalen2 * (
+                ka_mn2[0, 0, 0][ig, indm]
+                + minorfrac * (ka_mn2[0, 0, 0][ig, indmp] - ka_mn2[0, 0, 0][ig, indm])
+            )
+
+            taug[0, 0, 0][ig] = corradj * (
+                colamt[0, 0, 0][0]
+                * (
+                    fac00 * absa[0, 0, 0][ig, ind0]
+                    + fac10 * absa[0, 0, 0][ig, ind0p]
+                    + fac01 * absa[0, 0, 0][ig, ind1]
+                    + fac11 * absa[0, 0, 0][ig, ind1p]
+                )
+                + tauself
+                + taufor
+                + taun2
+            )
+
+            fracs[0, 0, 0][ig] = fracrefa[0, 0, 0][ig]
+
+    with computation(PARALLEL), interval(laytrop, nlay):
```

Comment:
I am not sure if it matters for performance or not, but you could merge this computation with the one above and have two 'with interval' statements. This is a common pattern, and you don't have to change it if you prefer not to.

Reply:
sounds good, will do